fabric-transaction-api-v1/src/main/java/net/fabricmc/fabric/api/transaction/v1/SimpleChangeListDataKey.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transaction.v1;
+
+import java.util.List;
+
+/**
+ * An implementation of {@link ChangeListDataKey} that provides prebuilt
+ * methods for applying changelists to the target object.
+ *
+ * @param <T> the change list item type
+ * @param <R> the target object type
+ */
+public interface SimpleChangeListDataKey<T, R> extends ChangeListDataKey<T> {
```

Comment:
I could actually just merge that together with `ChangeListDataKey` since I don't see a lot of use cases for the latter, and if it really is needed it can just be copy pasted.